stdlib/LinearAlgebra/test/symmetric.jl

```diff
@@ -365,6 +365,27 @@ end
                 @test Symmetric(asym)\b  ≈ asym\b
             end
         end
+
+        @testset "dot product of symmetric matrices" begin
+            symau = Symmetric(a, :U)
+            symal = Symmetric(a, :L)
+            msymau = Matrix(symau)
+            msymal = Matrix(symal)
+            @test_throws DimensionMismatch dot(symau, Symmetric(zeros(eltya, n-1, n-1)))
+            for eltyc in (Float32, Float64, ComplexF32, ComplexF64, BigFloat, Int)
+                creal = randn(n, n)/2
+                cimag = randn(n, n)/2
+                c = eltya == Int ? rand(1:7, n, n) : convert(Matrix{eltya}, eltya <: Complex ? complex.(creal, cimag) : creal)
+                symcu = Symmetric(c, :U)
+                symcl = Symmetric(c, :L)
+                msymcu = Matrix(symcu)
+                msymcl = Matrix(symcl)
+                @test dot(symau, symcu) ≈ dot(msymau, msymcu)
+                @test dot(symau, symcl) ≈ dot(msymau, msymcl)
+                @test dot(symal, symcu) ≈ dot(msymal, msymcu)
+                @test dot(symal, symcl) ≈ dot(msymal, msymcl)
+            end
+        end
```

Comment:
For the nested matrices, you could simply have a couple of tests like
```julia
X = [rand(eltyXYZ, 10, 10) for _ in 1:3, _ in 1:3]
@test dot(Symmetric(X), Symmetric(X)) ≈ dot(Matrix(Symmetric(X)), Matrix(Symmetric(X)))
```
There is no need to create block symmetric matrices by hand, simply let the wrapper do, and afterwards get rid of the wrapper by `Matrix`.